docs/sphinx/user_guide/using_camp.rst

```diff
@@ -0,0 +1,132 @@
+.. _using_camp-label: 
+
+**********
+Using Camp
+**********
+
+Since Camp is a metaprogramming library that is primarily header-only, it is hard
+to find isolated code examples that show Camp's capability. Instead, this page
+shows several real application examples of how Camp is being used currently.
+
+Camp Used in Umpire
+===================
+
+Umpire is a resource management library that allows the discovery, provision, and management of memory on machines 
+with multiple memory devices like NUMA and GPUs. Umpire is also part of the RAJA Portability Suite.
+Umpire's Operations provide an abstract interface to modifying and moving data between Umpire allocators.
+Camp makes these operations easy to port regardless of the underlying hardware. For example:
+
+.. code-block:: cpp
+
+  camp::resources::EventProxy<camp::resources::Resource> CudaMemsetOperation::apply_async(
```

Comment:
Note that keeping the code example in these docs consistent with the Umpire code will require manual care. Please consider linking in the actual code section here rather than cut-and-paste.

Reply:
The user-facing camp examples in Umpire are mostly just using resources (with the `wait()` and `get_event()` methods) which have already been shown in previous examples. This code snippet shows a bit more. I could look in the umpire tests or benchmarks which are typically a bit more complex than our super high-level dummy examples - stay tuned...

Reply:
There are a couple examples in Umpire of using literal includes to tell sphinx which lines of code to include into the .rst file... but that's all within Umpire, not linking to some other outside repo like it would be here between camp and umpire. I can try it and see if it works, but the main point in including these examples is just to demonstrate how camp is used in a realistic setting (working code) since Camp has zero examples - Therefore, I'm not sure it really matters if the true source code ends up changing a bit, since it would still be capturing a real, live, working (though out-of-date) camp code example. It's not the best solution, but again, no Camp examples